File: benchmarks/pb_ref_swap/mono.py
def f_swap(a_l, b_l):
    # python has no notion of references.
    # To get the desired effect, we must pass mutable args.
    tmp = a_l[0]
    a_l[0] = b_l[0]
    b_l[0] = tmp

def f_call(n):
    ct = 0;
    x1, x2 = [0], [0]

    for i in xrange(n):
        x1[0] = -i - n + 3
        x2[0] = i + n
        f_swap(x1, x2)
        ct += x1[0] + x2[0]

    # each loop iteration adds 3
    expect = n * 3
    assert ct == expect

def run_iter(n):
    f_call(n)

File: benchmarks/pb_ref_swap2/mono.php
<?php

function f_swap(&$a, &$b) {
	$tmp = $a;
	$a = $b;
	$b = $tmp;
}

function f_call($n) {

	$ct = 0;
	for ($i = 0; $i < $n; $i++) {
		$x1 = -$i - $n + 3;
		$x2 = $i + $n;
		f_swap($x1, $x2);
		$ct += $x1 + $x2;
	}
	// each loop iteration adds 3
	$expect = $n * 3;
	assert($ct == $expect);
}

function run_iter($n) {
	f_call($n);
}

?>
